src/bloqade/squin/wire.py

```diff
@@ -95,23 +95,18 @@ def __init__(self, operator: ir.SSAValue, *args: ir.SSAValue):
 class Measure(ir.Statement):
     traits = frozenset({lowering.FromPythonCall(), WireTerminator()})
     wire: ir.SSAValue = info.argument(WireType)
+    qubit: ir.SSAValue = info.argument(QubitType)
     result: ir.ResultValue = info.result(types.Int)
 
 
 @statement(dialect=dialect)
-class MeasureAndReset(ir.Statement):
-    traits = frozenset({lowering.FromPythonCall(), WireTerminator()})
-    wire: ir.SSAValue = info.argument(WireType)
+class NonDestructiveMeasure(ir.Statement):
+    traits = frozenset({lowering.FromPythonCall()})
+    input_wire: ir.SSAValue = info.argument(WireType)
```

Comment:
@cduck I think it would be good to understand the runtime behavior of this statement but @johnzl-777 please lets merge this PR and address it later. 

We can just error if you write this statement.